analyses/malawi/dryspells_trigger/exploration/compare_arc2_chirps.Rmd

```diff
@@ -0,0 +1,483 @@
+  mutate(trigger_year = year(chirps_confirmation)) %>%
+  group_by(trigger_year) %>%
+  summarize(n_triggers = n(),
+            .groups = "drop")
+```
+
+```{r trigger-center}
+arc2_center_df %>%
+  filter(month(arc2_center_confirmation) <= 2) %>%
+  mutate(trigger_year = year(arc2_center_confirmation)) %>%
+  group_by(trigger_year) %>%
+  summarize(n_triggers = n(),
+            .groups = "drop")
+```
+
+```{r trigger-touch}
+arc2_touch_df %>%
+  filter(month(arc2_touch_confirmation) <= 2) %>%
+  mutate(trigger_year = year(arc2_touch_confirmation)) %>%
+  group_by(trigger_year) %>%
+  summarize(n_triggers = n(),
+            .groups = "drop")
+```
+
+Given that the 2 ARC2 methods are so similar, we likely only need to look at the centroid method.
+This is because based on the matrix above, there are only 6 dry spells detected through
+the touch method not detected through the centroid method, but all but one of those
+is a city area. Alternatively, the centroid method would result in 8 triggers not captured
+by the touch method. However, only one of these is an urban area and the other 7
+are south-central districts in the southern region.
+
+```{r get-rid-of-touch}
+arc2_chirps_df %>%
+  filter(arc2_touch_ds, !arc2_centroid_ds) %>%
+  transmute(ADM2_EN, trigger_year = year(arc2_touch_confirmation))
+```
+
+```{r keep-that-centroid}
+arc2_chirps_df %>%
+  filter(!arc2_touch_ds, arc2_centroid_ds) %>%
+  transmute(ADM2_EN, trigger_year = year(arc2_center_confirmation))
+```
+
+Although the centroid method fails to capture Zomba City (due to a lack of raster
+centroid for ARC2 data), it seems sufficient to consider vs. CHIRPS excluding the
+touching raster method.
+
+### Daily value comparisons
+
+First, let's do some 2d histograms to examine the distribution of various calculation methods.
+What do we see between the centroid and touching raster methods applied?
+
+```{r 2d-histo-arc}
+daily_df %>%
+  ggplot(aes(x = arc2_centroid,
+             y = arc2_touch)) +
+  geom_bin2d(bins = 200) +
+  theme_bw() +
+  geom_abline(intercept = 0)
+```
+
+While it looks like the touching method might skew higher (if just slightly) at
+higher levels of rainfall, we are more concerned at their behavior at lower levels.
+
+```{r 2d-histo-arc-1}
+daily_df %>%
+  filter(arc2_centroid <= 1 | arc2_touch <= 1) %>%
+  ggplot(aes(x = arc2_centroid,
+             y = arc2_touch)) +
+  geom_bin2d(bins = 50) +
+  theme_bw() +
+  geom_abline(intercept = 0)
+```
+
+And above we can see that the centroid method in general trends much higher at these levels.
+This intuitively makes sense as the touching method takes the average of more raster
+cells than the centroid method, so will tend to regress toward the mean at the extremes.
+Anyway, given their similar results in identifying dry spells and the results above,
+should be fine to simply compare the centroid method against CHIRPS and not the
+touching method.
+
+```{r 2d-histo-chirps}
+daily_df %>%
```

Comment:
Not critical, but what does it look like if you colour the pixels according to a log scale? Just a bit difficult to see differences in the z axis since it all looks the same colour.

Reply:
Yeah good call.